Firestore/core/src/core/filter.h

```diff
@@ -23,10 +23,12 @@
 #include <vector>
 
 #include "Firestore/core/src/model/model_fwd.h"
+#include "Firestore/core/src/util/thread_safe_memoizer.h"
+
+using firebase::firestore::util::ThreadSafeMemoizer;
```

Comment:
nit: Remove this `using` statement from `filter.h`. In general, don't use `using` statement in header files. The reason is that now anyone who includes `filter.h` will see `ThreadSafeMemozier` as part of the global namespace. In `.cc` files this is fine because nobody _includes_ .cc files. Note that instead of the `using` statement, you can just use `util::ThreadSafeMemoizer` since the `firebase::firestore::` prefix is implied for code that exists in the `firebase::firestore::core` namespace.